packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialogWrapper.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Tupaia Config Server
+ * Copyright (c) 2019 Beyond Essential Systems Pty Ltd
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { closeEnlargedDialog } from '../../actions';
+import { selectIsEnlargedDialogVisible } from '../../selectors';
+import { EnlargedDialog } from './EnlargedDialog';
+
+const EnlargedDialogWrapperComponent = ({ isVisible, onCloseOverlay }) => {
```

Comment:
hmmm I'm not sure about this component - it seems a bit unnecessary. What do you think about moving the onCloseOverlay down to EnlargedDialog and moving the isVisible up to the Screen components?

Reply:
The other option would be to reset all the state to default values in the EnlargedDialog component but that could get messy. One potential solution for that is to use the `useReducer` hook. I'm not sure if you have seen that but it is good when there are a lot of interacting uses of `useState` in a component. I think it might be going too far to use that here but it is good to know about.

Reply:
> What do you think about moving the onCloseOverlay down to EnlargedDialog and moving the isVisible up to the Screen components?

I've gone with this approach.

> One potential solution for that is to use the useReducer hook

I hadn't seen it before but it looks interesting! Will definitely keep it in mind for the future.